fastapi/encoders.py

```diff
@@ -15,13 +16,11 @@
 def generate_encoders_by_class_tuples(
     type_encoder_map: Dict[Any, Callable]
 ) -> Dict[Callable, Tuple]:
-    encoders_by_classes: Dict[Callable, List] = {}
+    encoders_by_classes: Dict[Callable, List] = defaultdict(list)
```

Comment:
remove also `encoders_by_classes[encoder].append(type_)` in line 21 cause the suggeriment above

Reply:
It won't actually work, because `encoder` should be a key of a dictionary and `type_` should be a tuple of classes.

But the main problem that one `encoder` can have multiple `type_` and such case can't be handled using comprehension.

For instance result of an original function call will look like:
```python
{
   <cyfunction <lambda> at 0x7ff7f278e5c0>: (<class 'bytes'>,), 
   <class 'str'>: (<class 'pydantic.color.Color'>, <class 'ipaddress.IPv4Address'>, <class 'ipaddress.IPv4Interface'>, <class 'ipaddress.IPv4Network'>, <class 'ipaddress.IPv6Address'>, <class 'ipaddress.IPv6Interface'>, <class 'ipaddress.IPv6Network'>, <class 'pathlib.Path'>, <class 'pydantic.types.SecretBytes'>, <class 'pydantic.types.SecretStr'>, <class 'uuid.UUID'>), 
   <cyfunction isoformat at 0x7ff7f278e4f8>: (<class 'datetime.date'>, <class 'datetime.datetime'>, <class 'datetime.time'>), 
   <cyfunction <lambda> at 0x7ff7f278e688>: (<class 'datetime.timedelta'>,), 
   <class 'float'>: (<class 'decimal.Decimal'>,), 
   <cyfunction <lambda> at 0x7ff7f278e750>: (<enum 'Enum'>,), 
   <class 'list'>: (<class 'frozenset'>, <class 'generator'>, <class 'set'>)
}
```
In case when we use comprehension:
```python
{
   <cyfunction <lambda> at 0x7f8f8b0775c0>: (<class 'bytes'>,),
   <class 'str'>: (<class 'uuid.UUID'>,), 
   <cyfunction isoformat at 0x7f8f8b0774f8>: (<class 'datetime.time'>,), 
   <cyfunction <lambda> at 0x7f8f8b077688>: (<class 'datetime.timedelta'>,), 
   <class 'float'>: (<class 'decimal.Decimal'>,), 
   <cyfunction <lambda> at 0x7f8f8b077750>: (<enum 'Enum'>,), 
   <class 'list'>: (<class 'set'>,)
}
```